sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/resolution/ExpressionResolver.kt

```diff
@@ -111,4 +122,40 @@ private fun resolve(literalValue: SqliteParser.Literal_valueContext): Resolution
     return Resolution(values = listOf(Value(null, null, Value.SqliteType.NULL, literalValue, null)))
   }
   return Resolution(ResolutionError.IncompleteRule(literalValue, "Unsupported literal"))
+}
+
+private fun Resolver.resolveFunction(
+    expression: SqliteParser.ExprContext,
+    subqueriesAllowed: Boolean
+): Resolution {
+  var sqliteType = Value.SqliteType.INTEGER
+  val resolutions = expression.expr().map { resolve(it, subqueriesAllowed) }
+  expression.function_name().text.let { functionName ->
+    when (functionName) {
+      "date", "time", "datetime", "julianday", "strftime", "char", "hex", "lower", "ltrim",
+      "printf", "quote", "replace", "rtrim", "soundex", "sqlite_compileoption_get",
+      "sqlite_source_id", "sqlite_version", "substr", "trim", "typeof", "upper",
+      "group_concat" -> {
+        sqliteType = Value.SqliteType.TEXT
+      }
+      "changes", "instr", "last_insert_rowid", "length", "random", "sqlite_compileoption_used",
+      "total_changes", "unicode", "count" -> {
+        sqliteType = Value.SqliteType.INTEGER
+      }
+      "randomblob", "zeroblob" -> {
+        sqliteType = Value.SqliteType.BLOB
+      }
+      "round", "avg", "sum", "total" -> {
```

Comment:
add a todo? summing an int column and being forced to use a real in a query mapper (since presumably we'll use the same type resolution) would be odd.